core/services/chainlink/config_capabilities.go

```diff
@@ -42,3 +58,18 @@ func (c *capabilitiesExternalRegistry) ChainID() string {
 func (c *capabilitiesExternalRegistry) Address() string {
 	return *c.c.Address
 }
+
+type workflowConnectorConfig struct {
+	c toml.WorkflowConnectorConfig
+}
+
+func (c *workflowConnectorConfig) ChainIDForNodeKey() big.Int {
+	return c.c.ChainIDForNodeKey
+}
+
+func (c *workflowConnectorConfig) GatewayConnectorConfig() toml.GatewayConnectorConfig {
+	// invalid operation: cannot indirect
+	// c.c.GatewayConnectorConfig (variable of type "github.com/smartcontractkit/chainlink/v2/core/config/toml".ConnectorConfig)
+	// compilerInvalidIndirection
+	return c.c.GatewayConnectorConfig
```

Comment:
The toml struct won't be its own satisfy GatewayConnectorConfig since it doesn't have any of the interface methods on it. You'll need to define a wrapper like this and do the mapping of method on the interface to the attribute on the underlying toml fragment yourself.

Reply:
ty.  Having read your comment, now the error seems perfectly clear.